src/crush/CrushWrapper.cc

```diff
@@ -3131,7 +3131,21 @@ class CrushTreeFormattingDumper : public CrushTreeDumper::FormattingDumper {
     f->open_array_section("nodes");
     Parent::dump(f);
     f->close_section();
+
+    // There is no stray bucket whose id is a negative number, so just get
+    // the max_id and iterate from 0 to max_id to dump stray osds.
     f->open_array_section("stray");
+    auto &name_map = this->crush->name_map;
```

Comment:
nit, i'd suggest avoid introducing another variable for `this->crush->name_map`, as `crush->name_map` is short/good enough.